pkgs/development/beam-modules/fetch-mix-deps.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenvNoCC, elixir, hex, rebar, rebar3, cacert, git }:
+
+{ name, version, sha256, src, mixEnv ? "prod", debug ? false, meta ? { } }:
```

Comment:
This confuses me: Are there to inputs?

Reply:
I think you meant 'two' inputs. If you did here is what I think.
This confuses me too. I checked mostly how fetch rebar was implemented.
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/beam-modules/fetch-rebar-deps.nix
The rationale I think is that you want to be able to make several versions of the function.
For example, there is a problem with rebar3 version 3.14 and elixir 1.11
You can make a version fetchMix and a fetchMixRebar3_14 or something.
This is how I understood it. I'm open to suggestions.

Reply:
The first set of arguments is for the "fetcher" and the second is for the dependency set to be fetched. This is the same pattern used by every other builder and fetcher expression in beam-modules.